README.md

```diff
@@ -56,10 +56,34 @@ set laststatus = 0
 ```
 you can of course not use `BufEnter` and use `WinEnter` or some other events but I found it sufficient for my everyday use.
 
-## The api
+## The Internals
+The plugin's implementation is simple, it just runs `tmux source 'the content of your json file'`, everything else is just deserialization and formatting.
+
+Let us write an example with the current file type on the left.
+In your onestatus.json;
+```json
+  "status-left": [
+    { "fg": "default", "bg": "default", "labelFunc": "g:CurFileType" }
+  ]
+```
+And make a function in your `$MYVIMRC` to return the file type.
+```vim
+fun CurFileType()
```

Comment:
since file type means something special in vim (:h filetype), `CurrentFileExtension` should be more adequate